multicluster/apis/multicluster/v1alpha1/clusterset_types.go

```diff
@@ -47,10 +45,73 @@ type ClusterSetSpec struct {
 	Namespace string `json:"namespace,omitempty"`
 }
 
-// ClusterSetStatus defines the observed state of ClusterSet
+type ClusterSetConditionType string
+
+const (
+	// ClusterSetReady indicates whether ClusterSet is ready.
+	ClusterSetReady ClusterSetConditionType = "ClusterSetReady"
+)
+
+// ClusterSetCondition indicates the readiness condition of the clusterSet.
+type ClusterSetCondition struct {
+	Type ClusterSetConditionType `json:"type,omitempty"`
+	// Status of the condition, one of True, False, Unknown.
+	Status v1.ConditionStatus `json:"status,omitempty"`
+	// +optional
+	// Last time the condition transit from one status to another.
+	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty"`
+	// +optional
+	// A human readable message indicating details about the transition.
+	Message string `json:"message,omitempty"`
+	// +optional
+	// Unique, one-word, CamelCase reason for the condition's last transition.
+	Reason string `json:"reason,omitempty"`
+}
+
+type ClusterConditionType string
+
+const (
+	// ClusterReady indicates whether Cluster is ready and connected.
+	ClusterReady ClusterConditionType = "ClusterReady"
+	// ClusterIsLeader indicates whether Cluster is leader.
+	ClusterIsLeader ClusterConditionType = "ClusterIsLeader"
+)
+
+// ClusterCondition indicates the readiness condition of a cluster.
+type ClusterCondition struct {
+	Type ClusterConditionType `json:"type,omitempty"`
+	// Status of the condition, one of True, False, Unknown.
+	Status v1.ConditionStatus `json:"status,omitempty"`
+	// ClusterID is the unique identifier of this cluster.
+	ClusterID string `json:"clusterID,omitempty"`
+
+	// +optional
+	// Last time the condition transit from one status to another.
+	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty"`
+	// +optional
+	// A human readable message indicating details about the transition.
+	Message string `json:"message,omitempty"`
+	// +optional
+	// Unique, one-word, CamelCase reason for the condition's last transition.
+	Reason string `json:"reason,omitempty"`
+}
+
+type ClusterStatus struct {
+	Conditions []ClusterCondition `json:"conditions,omitempty"`
+}
+
+// ClusterSetStatus defines the observed state of ClusterSet.
 type ClusterSetStatus struct {
-	// Important: Run "make" to regenerate code after modifying this file
-	// TBD
+	// Total number of member clusters configured in the set.
+	TotalClusters int32 `json:"totalClusters,omitempty"`
```

Comment:
Could we calculate from member clusters and their status? Maybe we just like a more convenient way?